### PR TITLE
Add config for building capi 1.0.0 into hypershift namespace

### PR DESCRIPTION
**Pull request by enxebre**
This is to build and publish [CAPI release-1.0](https://github.com/openshift/cluster-api/tree/release-1.0) into the hypershift namespace so it can be consumed from hypershift as registry.ci.openshift.org/hypershift/cluster-api:v1.0.0.

It copies the caconical CAPI Docker file but sets ENV GOFLAGS="-mod=readonly".

### CODE REVIEW

**Comment by enxebre** [comment]
/hold
to set up docker build properly.

File: ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml
Comment (ironcladlou) [inline ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml:19]: Should this be `registry.ci.openshift.org/openshift/release:golang-1.16`?

File: ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml
Comment (ironcladlou) [inline ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml:32]: Not clear to me how any of these are used

File: ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml
Comment (enxebre) [reply ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml:19]: I set the current one because otherwise the built failed with some go vendor issue, I assume because the config is different. Not sure it makes a big difference since then the dockerfile use another image to contain the final binary https://github.com/openshift/cluster-api/blob/cluster-api-0.1.0/Dockerfile#L16-L28

I was planning to get this in as first step then follow up to swap to the right images and maybe change the dockerfile but already driven CI

File: ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml
Comment (enxebre) [reply ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml]: Actually neither is clear to me, dropped it.

File: ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml
Comment (alvaroaleman) [reply ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml:19]: This will end up flaking because docker.io pulls rate ratelimited. IIRC the image dan linked sets the `GOFLAGS=-mod=vendor` env var or sth like that which could explain the issues you are seeing

**Review by alvaroaleman — COMMENTED** [review]
I think we have to find a way to get rid of the docker.io image due to ratelimiting, other than that it looks good

File: ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml
Comment (enxebre) [reply ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml]: Right https://github.com/openshift/release/blob/c32756180704c239f76063c8d39bc464ea7d2374[…]ters/01_cluster/ci/_origin-release-build/golang-1.16/Dockerfile

I updated to use the right image an line the Docker file to set the Env variable.

**Review by alvaroaleman — APPROVED** [review]
thanks!

File: ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml
Comment (ironcladlou) [inline ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml:59]: Should this be `quay.io/openshift/origin-base:4.9`?

Comment (enxebre) [reply ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml:59]: My understanding is that it's overiden by https://github.com/openshift/release/pull/23206/files#diff-2d44cebb87ecd8b08ae33b42c62ba27f714d30ee2c61086341f71145d65e2b4fR65 but I can set it explicitly as well.

Comment (enxebre) [reply ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml]: done.

Comment (ironcladlou) [reply ci-operator/config/openshift/cluster-api/openshift-cluster-api-release-1.0.yaml]: Magic!

**Comment by ironcladlou** [comment]
Everything I noticed has been addressed, will leave a hold for others to remove when they're satisfied. Thanks!

/lgtm
/hold

**Comment by ironcladlou** [comment]
Looks like I'm not an approver here anyway 😁

**Comment by enxebre** [comment]
Thanks! let's proceed and watch the image get built.
/hold cancel

**Pull request merged**
